storage-proofs-update/src/constants.rs

```diff
@@ -2,23 +2,48 @@ use blstrs::Scalar as Fr;
 use filecoin_hashers::{
     poseidon::{PoseidonDomain, PoseidonHasher},
     sha256::{Sha256Domain, Sha256Hasher},
+    FieldArity, PoseidonArity,
 };
-use generic_array::typenum::{Unsigned, U0, U2, U8};
+use generic_array::typenum::{U0, U2, U8};
 use lazy_static::lazy_static;
 use merkletree::store::DiskStore;
 use neptune::{
     hash_type::{CType, HashType},
     poseidon::PoseidonConstants,
     Strength,
 };
-use storage_proofs_core::merkle::{BinaryMerkleTree, LCStore, LCTree, MerkleTreeTrait};
-
-// Use a custom domain separation tag when generating randomness phi, rho, and challenges bits.
-pub const HASH_TYPE_GEN_RANDOMNESS: HashType<Fr, U2> = HashType::Custom(CType::Arbitrary(1));
+use pasta_curves::{Fp, Fq};
+use storage_proofs_core::merkle::{BinaryMerkleTree, LCTree};
+use typemap::ShareMap;
 
 lazy_static! {
-    pub static ref POSEIDON_CONSTANTS_GEN_RANDOMNESS: PoseidonConstants::<Fr, U2> =
-        PoseidonConstants::new_with_strength_and_type(Strength::Standard, HASH_TYPE_GEN_RANDOMNESS);
+    // Use a custom domain separation tag `HashType` when using Poseidon to generate randomness
+    // (i.e. phi, rho, and challenges bits).
+    pub static ref POSEIDON_CONSTANTS_GEN_RANDOMNESS_BLS: PoseidonConstants<Fr, U2> =
```

Comment:
Note: create Poseidon constants which use the "randomness generation" DST for the three fields; store in a `typemap` so that we can lookup the constants using `(field, artity)`.